### ISSUE TITLE
Grounder returns non-zero exit status

### ISSUE THREAD

**Issue by basus** [ocaml/opam#3484]
I've been running into the following error whenever I try to run `opam update` or `opam upgrade`. 

```

[WARNING] External solver failed:
# opam-version 1.2.2
# os           darwin
# command      aspcud /Users/basus/.opam/log/solver-in-25652-7afd23 /Users/basus/.opam/log/solver-out-25652-5f1023 -count(down),-count(removed),-notuptodate(solution),-sum(solution,version-lag),-count(new)
# path         /Users/basus
# exit-code    1
# env-file     /Users/basus/.opam/log/log-25652-58c514.env
# stdout-file  /Users/basus/.opam/log/log-25652-58c514.out
# stderr-file  /Users/basus/.opam/log/log-25652-58c514.err
### stderr ###
# ERROR: grounder returned with non-zero exit status

[ERROR] External solver failure, please fix your installation and check /Users/basus/.opam/config and variable $OPAMEXTERNALSOLVER.
        You may also retry with option --use-internal-solver
```

Here are the contents of the `.env` file referenced above. The `.out` file is empty and the `.err` file just has the above ERROR message:

```
Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.vKd0BzICzb/Render
CAML_LD_LIBRARY_PATH=/Users/basus/.opam/4.06.1/lib/stublibs
COLORFGBG=15;0
COLORTERM=truecolor
COMMAND_MODE=unix2003
EDITOR=emacsclient
GEM_HOME=/Users/basus/.rvm/gems/ruby-2.3.0
GEM_PATH=/Users/basus/.rvm/gems/ruby-2.3.0:/Users/basus/.rvm/gems/ruby-2.3.0@global
HOME=/Users/basus
IRBRC=/Users/basus/.rvm/rubies/ruby-2.3.0/.irbrc
ITERM_PROFILE=Zenburn
ITERM_SESSION_ID=w0t0p0:BF7A8B0C-BFD9-486E-B39C-D4074CC248FE
JAVA_HOME=/usr/lib/jvm/java-8-openjdk
LANG=en_US.UTF-8
LC_CTYPE=en_US.UTF-8
LESS=-R
LOGNAME=basus
LSCOLORS=Gxfxcxdxbxegedabagacad
MANPATH=/Users/basus/.opam/4.06.1/man:/usr/share/man:/usr/local/share/man:/Library/TeX/Distributions/.DefaultTeX/Contents/Man
MY_RUBY_HOME=/Users/basus/.rvm/rubies/ruby-2.3.0
OCAML_TOPLEVEL_PATH=/Users/basus/.opam/4.06.1/lib/toplevel
OLDPWD=/Users/basus
OPAMUTF8MSGS=1
PAGER=less
PATH=/Users/basus/.rvm/gems/ruby-2.3.0/bin:/Users/basus/.rvm/gems/ruby-2.3.0@global/bin:/Users/basus/.rvm/rubies/ruby-2.3.0/bin:/Users/basus/.opam/4.06.1/bin:/usr/local/opt/coreutils/libexec/gnubin:/Users/basus/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/jvm/java-8-openjdk/bin:/Users/basus/.cargo/bin:/Applications/Racket v6.4/bin:/usr/local/texlive/2017/bin/x86_64-darwin:/Users/basus/.rvm/bin
PERL5LIB=/Users/basus/.opam/4.06.1/lib/perl5
PWD=/Users/basus
RUBY_VERSION=ruby-2.3.0
SECURITYSESSIONID=186a7
SHELL=/bin/zsh
SHLVL=2
SSH_AGENT_PID=6387
SSH_AUTH_SOCK=/var/folders/px/knyrxd_d3wb4220y4j7cmkg40000gn/T//ssh-SM1Re5bdGNo5/agent.6386
TERM=screen-256color
TERM_PROGRAM=iTerm.app
TERM_PROGRAM_VERSION=3.2.0beta8
TERM_SESSION_ID=w0t0p0:BF7A8B0C-BFD9-486E-B39C-D4074CC248FE
TMPDIR=/var/folders/px/knyrxd_d3wb4220y4j7cmkg40000gn/T/
TMUX=/private/tmp/tmux-501/default,6467,0
TMUX_PANE=%2
USER=basus
XPC_FLAGS=0x0
XPC_SERVICE_NAME=0
ZSH_TMUX_TERM=screen-256color
_ZSH_TMUX_FIXED_CONFIG=/Users/basus/.oh-my-zsh/plugins/tmux/tmux.extra.conf
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
rvm_bin_path=/Users/basus/.rvm/bin
rvm_path=/Users/basus/.rvm
rvm_prefix=/Users/basus
rvm_version=1.27.0 (latest)
LD_LIBRARY_PATH=:/usr/local/lib
_system_type=Darwin
_system_name=OSX
_system_version=10.13
_system_arch=x86_64
_=/usr/local/bin/opam

```

Here is the result of running `opam config report`

```
# OPAM config report
# opam-version    1.2.2 
# self-upgrade    no
# os              darwin
# external-solver aspcud $in $out $criteria
# criteria        -count(removed),-notuptodate(request),-sum(request,version-lag),-count(down),-notuptodate(changed),-count(changed),-notuptodate(solution),-sum(solution,version-lag)
# jobs            4
# repositories    1* (http)
# pinned          0
# current-switch  4.06.1
# last-update     2018-07-26 18:12
```

**Comment by AltGr** [comment]
Something seems broken with your installation of `aspcud`. You could try [updating to 2.0.0](http://www.ocamlpro.com/2018/07/26/opam-2-0-0-rc4-final-is-out/), which doesn't need it anymore.

**Comment by basus** [comment]
Turns out I previously linked my system Python from 3.7 to 3.6. That caused `clingo` to break, which in turn caused `aspcud` to break, which in turn broke OPAM.

**Comment by ilmoro90** [comment]
how did you fix the error?

`[ERROR] Solver failed: "/usr/bin/aspcud /root/.opam/log/solver-in-2562-5f1023
        /root/.opam/log/solver-out-2562-58c514
        -count(removed),-sum(request,version-lag),-count(down),-sum(solution,version-lag),-count(changed)"
        exited with code 1 "ERROR: grounder returned with non-zero exit status"
Switch initialisation failed: clean up? ('n' will leave the switch partially installed) [Y/n] `

**Comment by basus** [comment]
@ilmoro90 I suggest checking your Python setup and making sure that you are using Python 3.7. Unfortunately I don't remember anything else about how I fixed it.

**Comment by Dean177** [comment]
@ilmoro90 
For anyone else that runs into this, the fix for me was found in #3827:
```sh
rm -rf ~/.opam
apt install mccs
opam init --solver=mccs
opam switch create 4.10.0 --solver=mccs
```
